### PR TITLE
Add 'policy_document' to target.

### PR DESCRIPTION
**Pull request by thbishop-intuit**

### CODE REVIEW

File: internal/requests/requests.go [hunk 1]
```diff
@@ -7,8 +7,8 @@ import (
 	"github.com/argoproj-labs/argo-cloudops/internal/validations"
 )
 
+// CreateWorkflow request.
```
Comment (thbishop-intuit) [inline internal/requests/requests.go:10]: Fixed a number of public type/func comments.

File: service/internal/credentials/vault_test.go [hunk 2]
```diff
@@ -32,12 +32,6 @@ func TestVaultCreateProject(t *testing.T) {
 			admin:     false,
 			errResult: true,
 		},
-		{
```
Comment (thbishop-intuit) [inline service/internal/credentials/vault_test.go:35]: This was a duplicate test with the one below.

File: service/internal/credentials/vault_test.go [hunk 3]
```diff
@@ -253,12 +247,12 @@ func TestVaultGetTarget(t *testing.T) {
 			admin: true,
 		},
 		{
-			name:      "delete target admin error",
+			name:      "get target admin error",
```
Comment (thbishop-intuit) [inline service/internal/credentials/vault_test.go:250]: Typo

File: service/internal/credentials/vault_test.go [hunk 4]
```diff
@@ -253,12 +247,12 @@ func TestVaultGetTarget(t *testing.T) {
...
 			admin:     false,
 			errResult: true,
 		},
 		{
-			name:      "delete target error",
+			name:      "get target error",
```
Comment (thbishop-intuit) [inline service/internal/credentials/vault_test.go:255]: Fixed typo

File: service/internal/credentials/vault.go [hunk 5]
```diff
@@ -287,16 +289,31 @@ func (v VaultProvider) GetTarget(projectName, targetName string) (responses.Targ
...
-	policyArns := sec.Data["policy_arns"].([]interface{})
 	credentialType := sec.Data["credential_type"].(string)
 
-	var policies []string
-	for _, v := range policyArns {
-		policies = append(policies, v.(string))
+	// Optional.
+	policies := []string{}
+	val, ok := sec.Data["policy_arns"]
+	if ok {
```
Comment (Acepie) [inline service/internal/credentials/vault.go:299]: nit: any reason this one is 2 separate lines vs the one below?

**Review by Acepie — DISMISSED** [review]
1 tiny nit otherwise LGTM

**Review by pclata — DISMISSED** [review]
LGTM

File: service/internal/credentials/vault.go (hunk 5 above)
Comment (thbishop-intuit) [reply service/internal/credentials/vault.go:299]: ah, yep. was a refactor. will update

**Review by pclata — APPROVED** [review]

**Review by Acepie — APPROVED** [review]
LGTM

**Review by jk-intuit — APPROVED** [review]
LGTM

**Pull request merged**